src/Parser/Core/Modules/StatTracker.js

```diff
@@ -0,0 +1,335 @@
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SPECS from 'common/SPECS';
+import { calculateSecondaryStatDefault } from 'common/stats';
+import { formatMilliseconds } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+const debug = true;
+
+// TODO: stat constants somewhere else? they're largely copied from combatant
+class StatTracker extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  static STAT_BUFFS = {
+    //// POTIONS ////
+    [SPELLS.POTION_OF_PROLONGED_POWER.id]: { str: 2500, agi: 2500, int: 2500 },
+    // probably no need to add flasks/food because they never fall
+
+    //// TRINKETS ////
+    [SPELLS.SHADOWS_STRIKE.id]: {
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      crit: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+    [SPELLS.SHADOW_MASTER.id]: {
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      mastery: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+    [SPELLS.SWARMING_SHADOWS.id]: {
+      itemId: ITEMS.DREADSTONE_OF_ENDLESS_SHADOWS.id,
+      haste: (_, item) => calculateSecondaryStatDefault(845, 3480, item.itemLevel),
+    },
+    [SPELLS.RISING_TIDES.id]: { // TODO this trinket stacks oddly, results won't be quite right
+      itemId: ITEMS.CHARM_OF_THE_RISING_TIDE.id,
+      haste: (_, item) => calculateSecondaryStatDefault(900, 576, item.itemLevel),
+    },
+
+    //// MISC ////
+    [SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_STRENGTH.id]: { // check numbers
+      str: combatant => 3000 + 200 * (combatant.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id]),
+    },
+    [SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_AGILITY.id]: { // check numbers
+      agi: combatant => 3000 + 200 * (combatant.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id]),
+    },
+    [SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_INTELLECT.id]: { // check numbers
+      int: combatant => 3000 + 200 * (combatant.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id]),
+    },
+    [SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_VERSATILITY.id]: { // check numbers
+      vers: combatant => 1500 + 100 * (combatant.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id]),
+    },
+    [SPELLS.JACINS_RUSE.id]: { mastery: 3000 },
+    [SPELLS.MARK_OF_THE_CLAW.id]: { crit: 1000, haste: 1000 },
+
+    //// DRUID ////
+    [SPELLS.ASTRAL_HARMONY.id]: { mastery: 4000 },
+
+    //// PRIEST ////
+    [SPELLS.MIND_QUICKENING.id]: { haste: 800 },
+
+
+  }
+
+  _startingStats = {};
+  _stats = {};
+
+  on_initialized() {
+    this._startingStats = {
+      str: this.combatants.selected.strength,
```

Comment:
ok you'll think this is annoying, but please use the full name like in combatants; strength, agility, etc

Reply:
yeah yeah yeah